src/components/tables/AlertHistory/index.tsx

```diff
@@ -91,39 +111,60 @@ function AlertHistoryTable({
     const [beforeCursor, setBeforeCursor] = useState<string | undefined>(
         undefined
     );
-    const [afterCursor, setAfterCursor] = useState<string | undefined>(
-        undefined
+    const [cursorHistory, setCursorHistory] = useState<(string | undefined)[]>(
+        []
```

Comment:
The `AlertHistory` pagination does not have `after` and `first` so keeping track of the cursors so we can pagination both ways (some what). There are some limitations if the resolved alerts change after we have these cached.